core/src/main/java/org/infinispan/factories/InterceptorChainFactory.java

```diff
@@ -125,9 +124,10 @@ private AsyncInterceptorChain buildInterceptorChain() {
       }
       interceptorChain.appendInterceptor(createInterceptor(new InvocationContextInterceptor(), InvocationContextInterceptor.class), false);
 
-      if (!configuration.transaction().transactionMode().isTransactional()) {
-         interceptorChain.appendInterceptor(createInterceptor(new VersionInterceptor(), VersionInterceptor.class), false);
-      }
+      //TODO! why adding version for replace?
+      //if (!configuration.transaction().transactionMode().isTransactional()) {
+      //   interceptorChain.appendInterceptor(createInterceptor(new VersionInterceptor(), VersionInterceptor.class), false);
+      //}
```

Comment:
Is this still meant to be commented?

If we're removing this, should it be a separate commit/Jira?

Reply:
I forgot to ask @gustavonalle about it. Is the VersionInterceptor still needed? 
It sets some version for the replace commands and I didn't find any usage of that version. 
It breaks the xsite tests if non-tx cache (on site 1) backups to a tx cache (on site 2).

Reply:
I am not sure TBH. This is a relic from the compat mode times

Reply:
oh... I found a failure... EmbeddedRestMemcachedHotRodTest fails without it.
I need to drop the version in the receiver site... bah...